### ISSUE TITLE
Rule proposal: no unnecessary optional chaining 

### ISSUE THREAD

**Issue by mohsen1** [typescript-eslint/typescript-eslint#1641]
Since most of optional chaining today is compiled into a lot of code with `&&`, avoiding unnecessary optional chaining can be hugely beneficial. 

```ts
// bad
window?.document?.textContent

// good
window.document.textContent
```

**Comment by bradzacher** [comment]
This is already implemented! It's part of the `no-unnecessary-condition` rule:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-condition.md
